lib/ae_mdw/db/sync/stats.ex

```diff
@@ -0,0 +1,62 @@
+defmodule AeMdw.Db.Sync.Stats do
+  @moduledoc """
+  Update general and per contract stats during the syncing process.
+  """
+
+  alias AeMdw.Db.Model
+  alias AeMdw.Db.State
+  alias AeMdw.Stats
+  alias AeMdw.Util
+
+  require Model
+
+  @typep pubkey :: AeMdw.Node.Db.pubkey()
+
+  @spec update_nft_stats(State.t(), pubkey(), nil | pubkey(), pubkey()) :: State.t()
+  def update_nft_stats(state, contract_pk, prev_owner_pk, to_pk) do
+    state
+    |> increment_collection_nfts(contract_pk, prev_owner_pk)
+    |> decrement_collection_owners(contract_pk, prev_owner_pk)
+    |> increment_collection_owners(contract_pk, to_pk)
+  end
+
+  defp increment_collection_nfts(state, contract_pk, nil),
+    do: update_stat_counter(state, Stats.nfts_count_key(contract_pk))
+
+  defp increment_collection_nfts(state, _contract_pk, _prev_owner_pk), do: state
+
+  defp decrement_collection_owners(state, _contract_pk, nil), do: state
+
+  defp decrement_collection_owners(state, contract_pk, prev_owner_pk) do
+    case State.next(
+           state,
+           Model.NftOwnerToken,
+           {contract_pk, prev_owner_pk, -Util.max_256bit_int()}
+         ) do
+      {:ok, {^contract_pk, ^prev_owner_pk, _token}} ->
+        state
+
+      _new_owner ->
+        update_stat_counter(state, Stats.nft_owners_count_key(contract_pk), fn count ->
+          max(count - 1, 0)
+        end)
+    end
+  end
+
+  defp increment_collection_owners(state, contract_pk, to_pk) do
+    case State.next(state, Model.NftOwnerToken, {contract_pk, to_pk, -Util.max_256bit_int()}) do
+      {:ok, {^contract_pk, ^to_pk, _token}} -> state
+      _new_owner -> update_stat_counter(state, Stats.nft_owners_count_key(contract_pk))
```

Comment:
Since the `index` for `Model.Stat` is no longer `atom()`, can we update the type in the Model module?

Reply:
Done